src/components/medications-details-table.component.tsx

```diff
@@ -0,0 +1,284 @@
+        ]);
+      }, createErrorHandler);
+    };
+
+    const tableHeaders = [
+      { key: 'startDate', header: t('startDate', 'Start date'), isSortable: true, isVisible: true },
+      { key: 'details', header: t('details', 'Details'), isSortable: true, isVisible: true },
+    ];
+
+    const tableRows = currentMedicationPage.map((medication, id) => ({
+      id: `${id}`,
+      details: {
+        sortKey: medication.drug?.name,
+        content: (
+          <p className={styles.bodyLong01}>
+            <strong>{capitalize(medication.drug?.name)}</strong> &mdash; {medication.doseUnits?.display} &mdash;{' '}
+            {medication.route?.display}
+            <br />
+            <span className={styles.label01}>{t('dose', 'DOSE').toUpperCase()}</span>{' '}
+            <strong>{getDosage(medication.drug?.strength, medication.dose).toLowerCase()}</strong> &mdash;{' '}
+            {medication.frequency?.display} &mdash;{' '}
+            {t('medicationDurationAndUnit', 'for {duration} {durationUnit}', {
+              duration: medication.duration,
+              durationUnit: medication.durationUnits?.display,
+            })}
+            <br />
+            <span className={styles.label01}>{t('refills', 'REFILLS').toUpperCase()}</span> {medication.numRefills}
+          </p>
+        ),
+      },
+      startDate: {
+        sortKey: dayjs(medication.dateActivated).toDate(),
+        content: dayjs(medication.dateActivated).format('DD-MMM-YYYY'),
+      },
+    }));
+
+    const sortRow = (cellA, cellB, { sortDirection, sortStates }) => {
+      return sortDirection === sortStates.DESC
+        ? compare(cellB.sortKey, cellA.sortKey)
+        : compare(cellA.sortKey, cellB.sortKey);
+    };
+
+    return (
+      <DataTable headers={tableHeaders} rows={tableRows} isSortable={true} sortRow={sortRow}>
+        {({ rows, headers, getTableProps, getHeaderProps, getRowProps }) => (
+          <TableContainer title={title}>
+            {showAddNewButton && (
+              <TableToolbar>
+                <TableToolbarContent>
+                  <Button renderIcon={() => <Add16 />} onClick={() => alert('TODO')}>
+                    {' '}
+                    {/*TODO: Implement functionality*/}
+                    {t('add', 'Add')}
+                  </Button>
+                </TableToolbarContent>
+              </TableToolbar>
+            )}
+            <Table {...getTableProps()}>
+              <TableHead>
+                <TableRow>
+                  {headers.map(header => (
+                    <TableHeader
+                      {...getHeaderProps({
+                        header,
+                        isSortable: header.isSortable,
+                      })}>
+                      {header.header}
+                    </TableHeader>
+                  ))}
+                  <TableHeader />
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {rows.map((row, rowIndex) => (
```

Comment:
This here leaves room for performance improvements.

We can discuss this tomorrow. Not necessary for this PR though.